### PR TITLE
Km sprint 3 dom spies

### PR DESCRIPTION
**Pull request by Kristiannmiller**
## Description
This PR:
- Adds Chai Spies tests to every dom manipulation method in the domUpdates.js file
- Adds needed parameters to the addToMyRecipes method on domUpdates.js
- Adds needed parameters to the addToFavorites function on scripts.js
- Renames some confusing method names in domUpdates.js

## Motivation and Context
This fulfills the requirement specified in the project rubric to Spy on all DOM manipulation methods to verify that they occurred and were called with the correct arguments.
It also fulfills the requirement in the project rubric to Use and afterEach hook to clean up spies in between tests.

## How Has This Been Tested?
Tests have been verified using the terminal and npm test. All tests pass.

## Types of changes
- [X] Bug fix (non-breaking change which fixes an issue)

## Checklist:
- [x] My code follows the code style of this project.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/domUpdates.js [hunk 1]
```diff
@@ -109,7 +110,7 @@ let domUpdates = {
     }).join(", ");
   },
 
-  generateInstructions(recipe, fullRecipeInfo) {
+  displayInstructions(recipe, fullRecipeInfo) {
```
Comment (ericberglund117) [inline src/domUpdates.js:113]: Nice. I like the changed name because now we know exactly what this DOM function does

File: test/user-test.js [hunk 2]
```diff
@@ -283,27 +283,32 @@ describe('User', function() {
...
+            "ingredient": 11547,
+            "amount": 5
+          },
+          {
+            "ingredient": 1082047,
+            "amount": 5
+          }];
+      user1.updateCurrentUserPantry(11477, 1)
+      user1.updateCurrentUserPantry(11297, -1)
+      expect(user1.pantry).to.deep.equal(expectedPantry);
```
Comment (ericberglund117) [inline test/user-test.js:311]: Totally forgot we didn't finish this last part of the test. Great catch!

File: test/domUpdates-test.js [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+    expect(domUpdates.addToMyRecipes).to.have.been.called(1);
+    expect(domUpdates.addToMyRecipes).to.have.been.called.with(mockEvent, currentUser, recipeData, ingredientsData);
+  })
+
+  it('should hide unchecked recipes', () => {
+    chai.spy.on(domUpdates, ['hideUncheckedRecipe'], () => {});
+
+    domUpdates.hideUncheckedRecipe(recipeData);
+
+    expect(domUpdates.hideUncheckedRecipe).to.have.been.called(1);
```
Comment (ericberglund117) [inline test/domUpdates-test.js:248]: Going through this, it seems like the spies essentially require the same test-style template. I like that. Makes things much easier! This is great and now we know FOR SURE that our DOM is updating as it should be

**Review by ericberglund117 — APPROVED** [review]

File: src/domUpdates.js (hunk 1 above)
Comment (chadrick-d-dev) [reply src/domUpdates.js:113]: I concur. Good change for what it does.

File: test/domUpdates-test.js (hunk 3 above)
Comment (chadrick-d-dev) [reply test/domUpdates-test.js:248]: Such a relief that it's actually more straight forward than I thought!

File: src/scripts.js [hunk 4]
```diff
@@ -34,12 +34,6 @@ let recipeData;
 window.addEventListener("load", checkData);
 document.addEventListener('click', (e) => modifyIngredientCount(e));
 document.addEventListener('click', (e) => submitPantryChanges(e));
-// window.addEventListener("load", createCards);
```
Comment (chadrick-d-dev) [inline src/scripts.js:37]: Good cleanup!

File: src/scripts.js [hunk 5]
```diff
@@ -153,8 +148,9 @@ function filterRecipes(filtered) {
 }
 
 // FAVORITE RECIPE FUNCTIONALITY
-function addToFavorites() {
-  domUpdates.addToMyRecipes(event, currentUser);
+function addToFavorites(event) {
+  console.log(event)
```
Comment (chadrick-d-dev) [inline src/scripts.js:152]: Let's get rid of these console logs when we circle back.

File: src/user.js [hunk 6]
```diff
@@ -46,7 +46,7 @@ class User {
       if(match === undefined) {
         return false
       }
-      recipeIng.quantity.amount <= match.amount
+      return recipeIng.quantity.amount <= match.amount
```
Comment (chadrick-d-dev) [inline src/user.js:49]: I didn't catch the naming here. We can look at the classes and naming when we refactor.

**Review by chadrick-d-dev — APPROVED** [review]

File: test/domUpdates-test.js (hunk 3 above)
Comment (Kristiannmiller) [reply test/domUpdates-test.js:248]: Thanks! I'd still really like to get in and do describe blocks for all of our test methods later on when we do the final polish and linter

**Pull request merged**